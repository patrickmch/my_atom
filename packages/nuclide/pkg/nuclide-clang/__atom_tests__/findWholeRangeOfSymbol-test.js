"use strict";

var _atom = require("atom");

function _findWholeRangeOfSymbol() {
  const data = _interopRequireDefault(require("../lib/findWholeRangeOfSymbol"));

  _findWholeRangeOfSymbol = function () {
    return data;
  };

  return data;
}

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the LICENSE file in
 * the root directory of this source tree.
 *
 * 
 * @format
 * @emails oncall+nuclide
 */
describe('findWholeRangeOfSymbol', () => {
  let editor = null;
  beforeEach(async () => {
    editor = await atom.workspace.open('sampleObjC.m');
  });
  it('finds the range of a non-selector symbol.', () => {
    const text = 'name';
    const spelling = 'name';
    const textRangeInSample = new _atom.Range([8, 31], [8, 35]);
    const extent = new _atom.Range([8, 31], [8, 35]);
    const ranges = (0, _findWholeRangeOfSymbol().default)(editor, text, textRangeInSample, spelling, extent); // The range returned should just be the range of the original text.

    expect(ranges).toMatchSnapshot();
  });
  it('finds the range of a fully qualified non-selector symbol.', () => {
    const text = 'name';
    const spelling = 'namespace::name';
    const textRangeInSample = new _atom.Range([8, 31], [8, 35]);
    const extent = new _atom.Range([8, 31], [8, 35]);
    const ranges = (0, _findWholeRangeOfSymbol().default)(editor, text, textRangeInSample, spelling, extent);
    expect(ranges).toMatchSnapshot();
  });
  it('finds the range of a selector with one argument.', () => {
    const text = 'cStringUsingEncoding';
    const spelling = 'cStringUsingEncoding:';
    const textRangeInSample = new _atom.Range([12, 39], [12, 59]);
    const extent = new _atom.Range([12, 35], [12, 83]);
    const ranges = (0, _findWholeRangeOfSymbol().default)(editor, text, textRangeInSample, spelling, extent);
    expect(ranges).toMatchSnapshot();
  });
  it('finds the range of a selector with multiple arguments, when any of the segments is the' + ' selected "text".', () => {
    const spelling = 'createDirectoryAtPath:withIntermediateDirectories:attributes:error:';

    const extent = _atom.Range.fromObject({
      start: {
        row: 17,
        column: 6
      },
      end: {
        row: 19,
        column: 56
      }
    });

    const text1 = 'createDirectoryAtPath';
    const textRangeInSample1 = new _atom.Range([17, 20], [17, 41]);
    const ranges1 = (0, _findWholeRangeOfSymbol().default)(editor, text1, textRangeInSample1, spelling, extent);
    expect(ranges1).toMatchSnapshot();
    const text2 = 'withIntermediateDirectories';
    const textRangeInSample2 = new _atom.Range([18, 14], [18, 41]);
    const ranges2 = (0, _findWholeRangeOfSymbol().default)(editor, text2, textRangeInSample2, spelling, extent);
    expect(ranges2).toMatchSnapshot();
    const text3 = 'attributes';
    const textRangeInSample3 = new _atom.Range([19, 31], [19, 41]);
    const ranges3 = (0, _findWholeRangeOfSymbol().default)(editor, text3, textRangeInSample3, spelling, extent);
    expect(ranges3).toMatchSnapshot();
    const text4 = 'createDirectoryAtPath';
    const textRangeInSample4 = new _atom.Range([19, 46], [19, 51]);
    const ranges4 = (0, _findWholeRangeOfSymbol().default)(editor, text4, textRangeInSample4, spelling, extent);
    expect(ranges4).toMatchSnapshot();
  });
});