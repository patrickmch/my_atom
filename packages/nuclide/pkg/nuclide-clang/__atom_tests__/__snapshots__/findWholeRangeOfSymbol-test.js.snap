// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findWholeRangeOfSymbol finds the range of a fully qualified non-selector symbol. 1`] = `
Array [
  Range {
    "end": Point {
      "column": 35,
      "row": 8,
    },
    "start": Point {
      "column": 31,
      "row": 8,
    },
  },
]
`;

exports[`findWholeRangeOfSymbol finds the range of a non-selector symbol. 1`] = `
Array [
  Range {
    "end": Point {
      "column": 35,
      "row": 8,
    },
    "start": Point {
      "column": 31,
      "row": 8,
    },
  },
]
`;

exports[`findWholeRangeOfSymbol finds the range of a selector with multiple arguments, when any of the segments is the selected "text". 1`] = `Array []`;

exports[`findWholeRangeOfSymbol finds the range of a selector with multiple arguments, when any of the segments is the selected "text". 2`] = `Array []`;

exports[`findWholeRangeOfSymbol finds the range of a selector with multiple arguments, when any of the segments is the selected "text". 3`] = `Array []`;

exports[`findWholeRangeOfSymbol finds the range of a selector with multiple arguments, when any of the segments is the selected "text". 4`] = `Array []`;

exports[`findWholeRangeOfSymbol finds the range of a selector with one argument. 1`] = `
Array [
  Range {
    "end": Point {
      "column": 60,
      "row": 12,
    },
    "start": Point {
      "column": 39,
      "row": 12,
    },
  },
]
`;
